### PR TITLE
Fix: avoid setting repo_path when a file is not in the repo

### LINKED ISSUE #726: Unable to determine Git repo path
I've been getting this error when I try to commit recently. 
I don't know how to reproduce, or what causes the error.
Tried to open/close sublime, with and without project. Nothing seems to help.

### PR DESCRIPTION
**Pull request by randy3k**
continue the discussion in #719 for not setting "git_savvy.repo_path" if the current file is not in the found repo.

close #726

-----

The maintainers of this repo require that all pull request submitters agree and adhere to the following:

- [x] I have read and agree to the [contribution guidelines](https://github.com/divmain/GitSavvy/blob/master/CONTRIBUTING.md).
 _(required)_
- [x] All related documentation has been updated to reflect the changes made. _(required)_
- [x] My commit messages are cleaned up and ready to merge. _(required)_

The maintainers of this repository require you to select the semantic version type that
the changes in this pull request represent.  Please select one of the following:
- [ ]  major
- [ ]  minor
- [x]  patch
- [ ]  documentation only

### CODE REVIEW

**Comment by randy3k** [comment]
@asfaltboy I have to switch back to `view.window()` at line [249](https://github.com/divmain/GitSavvy/blob/4404a05085a3f02ef08ca24784adbe4287334b84/core/git_command.py#L249-L249) because it causes error when switching windows quickly.

Instead, I have put a check [here](https://github.com/divmain/GitSavvy/blob/4404a05085a3f02ef08ca24784adbe4287334b84/core/git_command.py#L297-L297) to prevent throwing error when the view is detected.

Hope it works.

**Comment by randy3k** [comment]
@divmain Given `maintainerd` [bug](https://github.com/divmain/maintainerd/issues/1), may be`Require status checks to pass before merging` should be disabled  for now so that we can still merge the PR manually?

**Comment by asfaltboy** [comment]
Agreed!

**Comment by randy3k** [comment]
@asfaltboy could you test if my fix reproduces the "not found" error?

**Comment by asfaltboy** [comment]
I couldn't reproduce the original "panel pops with Unable to determine repo path", but what happens now is (rarely) the exception breaks through and I see:

```
`/usr/bin/git status -b --porcelain` failed with following output:

fatal: Not a git repository (or any of the parent directories): .git
```

Alongside the dialog suggesting to "Initialize a new git repo".

Just as a quick attempt at bisect, I went back to e51c47a and could not reproduce the issue. I assume this is related to a later change in `git_command`. @randy3k you probably know better than me which is the commit that introduced this; question: is it necessary? Can we just revert it?

**Comment by randy3k** [comment]
It is an attempt to allow editing a file out of a Git repo. Before these refactorings, if you do `subl ~/.bashrc`, it would prompt the initialization dialog. I think I would what is happening, most likely that the two window.active_view() gets different values when we switch views very quickly.

**Comment by randy3k** [comment]
We now only have pure Julia methods for A' * B, A * B' and A' * B'. Theoretically, it should be faster to use lapack routines if bandwidths is large (simulations suggested that lapack routines are better for bandwidth >100).

Other than that, it is working.

**Comment by randy3k** [comment]
The bug is more subtle than I originally thought. The commit e51c47a was not completely free of bug, it seems to work just because it takes no time to finish the call. If I add a little bit [delay](https://github.com/divmain/GitSavvy/blob/bd22c270041a79d109b7b882c89b718cf016e715/core/git_command.py#L248-L248), it will break [appveyor](https://ci.appveyor.com/project/divmain/gitsavvy).

We see a similar bug earlier because `git rev-parse --show-toplevel` takes some time to execute.

**Comment by randy3k** [comment]
I guess in order to fix this issue, GsUpdateStatusBarCommand needs something like

```python
with disable_throw_on_stderr():
      self.any_git_command(.....)
```

**Comment by asfaltboy** [comment]
@randy3k maybe due to it's (currently) rare occurrence, we should merge this PR and handle the error raised from `GsUpdateStatusBarCommand` (if we're certain that is the cause) in another PR ?

**Comment by randy3k** [comment]
I guess I have fixed the issue at `GsUpdateStatusBarCommand` by putting a `try` block there.

**Comment by randy3k** [comment]
and also by limiting `self.repo_path` to throw `ValueError` but not `GitSavvyError`.

**Comment by randy3k** [comment]
@asfaltboy please feel free to merge this if you are happy with it

File: core/git_command.py [hunk 1]
```diff
@@ -268,42 +270,41 @@ def find_git_toplevel(self, folder, throw_on_stderr):
...
+                # only set "git_savvy.repo_path" when the current file is in repo_path
+                if file_name and os.path.realpath(file_name).startswith(repo_path + os.path.sep):
+                    view.settings().set("git_savvy.repo_path", repo_path)
 
         return os.path.realpath(repo_path) if repo_path else repo_path
 
     @property
+    def short_repo_path(self):
+        if "HOME" in os.environ:
+            return self.repo_path.replace(os.environ["HOME"], "~")
```
Comment (asfaltboy) [inline core/git_command.py]: Is this necessary for some cases? I don't recall an  issue discussing it. Maybe, note the reason in a comment / docstring?

Comment (randy3k) [inline core/git_command.py]: I didn't add this. It was in line 274.

**Comment by randy3k** [comment]
Let's merge it first and fix the exception later.

**Comment by randy3k** [comment]
I have a quick look at the exception, it seems that there is no easy way to catch this if a dashboard is closed quickly. We need a way to gracefully terminate the rendering code of the dashboard.

**Pull request merged**